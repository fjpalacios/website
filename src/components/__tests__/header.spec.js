import * as React from 'react'
import Header from '../header'
import createComponentWithIntl from '../../../__helpers__/i18n-component'

describe('Header', () => {
  it('should render properly', () => {
    const header = createComponentWithIntl(<Header />)
    expect(header).toMatchSnapshot()
  })

  it('should show English language by default', () => {
    const { container } = createComponentWithIntl(<Header />, 'en')
    const subTitle = container.querySelectorAll('.header__subtitle')[0].textContent
    expect(subTitle).toBe('Software Developer')
  })

  it('should show Spanish language if required', () => {
    const { container } = createComponentWithIntl(<Header />)
    const subTitle = container.querySelectorAll('.header__subtitle')[0].textContent
    expect(subTitle).toBe('Desarrollador de software')
  })
})
