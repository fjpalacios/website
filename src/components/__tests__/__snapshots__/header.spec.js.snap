// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <h1
          class="header__title"
        >
          Francisco Javier Palacios Pérez
        </h1>
        <h2
          class="header__subtitle"
        >
          Desarrollador de software
        </h2>
        <div
          class="header__contact"
        >
          <div
            class="header__contact__item linkedin"
          >
            <a
              href="https://www.linkedin.com/in/fjpalacios/"
              rel="noreferrer"
              target="_blank"
            >
              <i
                class="icon-linkedin"
              />
              <p>
                fjpalacios
              </p>
            </a>
          </div>
          <div
            class="header__contact__item mail"
          >
            <a
              href="mailto:javi@fjp.es"
              rel="noreferrer"
              target="_blank"
            >
              <i
                class="icon-mail"
              />
              <p>
                javi@fjp.es
              </p>
            </a>
          </div>
          <div
            class="header__contact__item github-circled"
          >
            <a
              href="https://github.com/fjpalacios"
              rel="noreferrer"
              target="_blank"
            >
              <i
                class="icon-github-circled"
              />
              <p>
                fjpalacios
              </p>
            </a>
          </div>
          <div
            class="header__contact__item globe"
          >
            <a
              href="https://fjp.es"
              rel="noreferrer"
              target="_blank"
            >
              <i
                class="icon-globe"
              />
              <p>
                fjp.es
              </p>
            </a>
          </div>
          <div
            class="header__contact__item twitter"
          >
            <a
              href="https://twitter.com/fjpalacios"
              rel="noreferrer"
              target="_blank"
            >
              <i
                class="icon-twitter"
              />
              <p>
                fjpalacios
              </p>
            </a>
          </div>
          <div
            class="header__contact__item location"
          >
            <a
              href="https://osm.org/go/b_qODpV--"
              rel="noreferrer"
              target="_blank"
            >
              <i
                class="icon-location"
              />
              <p>
                Valencia, ES
              </p>
            </a>
          </div>
          <div
            class="header__contact__item phone"
          >
            <a
              href="tel:0034671995762"
              rel="noreferrer"
              target="_blank"
            >
              <i
                class="icon-phone"
              />
              <p>
                671995762
              </p>
            </a>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <h1
        class="header__title"
      >
        Francisco Javier Palacios Pérez
      </h1>
      <h2
        class="header__subtitle"
      >
        Desarrollador de software
      </h2>
      <div
        class="header__contact"
      >
        <div
          class="header__contact__item linkedin"
        >
          <a
            href="https://www.linkedin.com/in/fjpalacios/"
            rel="noreferrer"
            target="_blank"
          >
            <i
              class="icon-linkedin"
            />
            <p>
              fjpalacios
            </p>
          </a>
        </div>
        <div
          class="header__contact__item mail"
        >
          <a
            href="mailto:javi@fjp.es"
            rel="noreferrer"
            target="_blank"
          >
            <i
              class="icon-mail"
            />
            <p>
              javi@fjp.es
            </p>
          </a>
        </div>
        <div
          class="header__contact__item github-circled"
        >
          <a
            href="https://github.com/fjpalacios"
            rel="noreferrer"
            target="_blank"
          >
            <i
              class="icon-github-circled"
            />
            <p>
              fjpalacios
            </p>
          </a>
        </div>
        <div
          class="header__contact__item globe"
        >
          <a
            href="https://fjp.es"
            rel="noreferrer"
            target="_blank"
          >
            <i
              class="icon-globe"
            />
            <p>
              fjp.es
            </p>
          </a>
        </div>
        <div
          class="header__contact__item twitter"
        >
          <a
            href="https://twitter.com/fjpalacios"
            rel="noreferrer"
            target="_blank"
          >
            <i
              class="icon-twitter"
            />
            <p>
              fjpalacios
            </p>
          </a>
        </div>
        <div
          class="header__contact__item location"
        >
          <a
            href="https://osm.org/go/b_qODpV--"
            rel="noreferrer"
            target="_blank"
          >
            <i
              class="icon-location"
            />
            <p>
              Valencia, ES
            </p>
          </a>
        </div>
        <div
          class="header__contact__item phone"
        >
          <a
            href="tel:0034671995762"
            rel="noreferrer"
            target="_blank"
          >
            <i
              class="icon-phone"
            />
            <p>
              671995762
            </p>
          </a>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
